test/optim/test_optimize.py

```diff
@@ -1459,13 +1461,28 @@ def test_optimize_acqf_mixed_q1(self, mock_optimize_acqf):
                 num_restarts=num_restarts,
                 raw_samples=raw_samples,
                 options=options,
+                return_best_only=return_best_only,
                 post_processing_func=rounding_func,
             )
             # compute expected output
-            ff_acq_values = torch.stack(acq_val_rvs)
-            best = torch.argmax(ff_acq_values)
-            expected_candidates = candidate_rvs[best]
-            expected_acq_value = ff_acq_values[best]
+            best_acq_values = torch.tensor(
+                [torch.max(acq_values) for acq_values in acq_val_rvs]
+            ).to(acq_val_rvs[0])
```

Comment:
This shouldn't be needed here?
```suggestion
            )
```